src/services/cipher.service.ts

```diff
@@ -841,6 +846,19 @@ export class CipherService implements CipherServiceAbstraction {
         };
     }
 
+    setTopUsernames(decCiphers: CipherView[]) {
+        const logins = decCiphers.map(c => c.login.username);
+        const countedUsernames = Object.values(logins.reduce<Record<string, {login: string, occurrences: number}>>((obj, login) => {
+            if (obj[login] === undefined)
+               obj[login] = { login: login, occurrences: 1 };
+            else
+               obj[login].occurrences++;
+            return obj;
+         }, {}));
+
+        this.topUsernames = countedUsernames.sort((a, b) => b.occurrences - a.occurrences).slice(0, 5).map(x => x.login);
```

Comment:
After some testing, I think we could probably increase this to say 10 usernames - probably a bit more useful.